File: projecteuler/2.py
# By considering the terms in the Fibonacci sequence whose values do not exceed
#  four million, find the sum of the even-valued terms.

N = 4000000
a = 1
b = 2

total = 2
while b <= N:
	c = a + b

	a = b
	b = c

	if c % 2 == 0:
		total += c
print(total)

# Efficient method (after solved)
a = 0
b = 2
c = 0

total = 2
while c <= N:
	total += c

	c = 4*b + a

	a = b
	b = c
print(total)

File: projecteuler/2.rs
fn main() {
	let mut sum = 2;
	let mut a = 0;
	let mut b = 2;
	let mut c = 0;
	let n = 4000000;

	while c < n {
		sum += c;

		c = 4*b + a;
		a = b;
		b = c;
	}

	println!("Sum evens = {sum}");
}
